src/domain/enrolment/enrolmentForm/EnrolmentForm.tsx

```diff
@@ -57,15 +57,43 @@ const EnrolmentForm: React.FC<Props> = ({
         handleSubmit,
         touched,
         submitCount,
-        values: {
+        values,
+        setFieldValue,
+        handleChange,
+      }) => {
+        const {
           isSameResponsiblePerson,
           isMandatoryAdditionalInformationRequired,
-        },
-      }) => {
+          person: { phoneNumber },
+          studyGroup: {
+            person: { phoneNumber: studyGroupPhoneNumber },
+          },
+        } = values;
         const showErrorNotification = !isEmpty(errors) && !!submitCount;
         const errorLabelKeys = keyify(errors)
           .map((path) => nameToLabelPath[path])
           .filter((i) => i);
+        const hasPhoneNumber = () => {
+          return !!phoneNumber || !!studyGroupPhoneNumber;
+        };
+        const handlePersonPhoneNumberChange = (
+          e: React.ChangeEvent<HTMLInputElement>
+        ) => {
+          handleChange(e);
+          if (!e.target.value && !studyGroupPhoneNumber) {
+            setFieldValue('hasSmsNotification', false);
+          }
+        };
+
+        const handleStudyGroupPhoneNumberChange = (
+          e: React.ChangeEvent<HTMLInputElement>
+        ) => {
+          handleChange(e);
+          if (!e.target.value && !phoneNumber) {
+            setFieldValue('hasSmsNotification', false);
+          }
+        };
```

Comment:
This way of changing form value based on other form field might cause some problems. Maybe it happens to work without problems in this situation. Heres how creator of Formik recommends to do this:

https://github.com/formium/formik/issues/2204#issuecomment-574207100

Reply:
Some reason why the HDS checkbox checked value did not get cleared or at least the cleared checkbox was not rerendered after the value change. This was the only solution I get it to work. I tried to extend the HDS checkbox field component with a useEffect like that, but I think I used wrong hook for value reading / set.